### PR TITLE
Fix insertion/removal behavior

### PR DESCRIPTION
**Pull request by jaxter184**
PR for issue #62

### CODE REVIEW

File: src/node.rs [hunk 1]
```diff
@@ -680,4 +668,44 @@ mod test {
...
+        node.push("pos0");
+        node.insert("keyword", 6.0);
+        node.push("pos1");
+        assert_eq!(node.entries().len(), 3);
+
+        node.insert(0, "inserted0");
+        node.insert(1, "inserted1");
+        assert_eq!(node.entries().len(), 3);
+        assert_eq!(node[0], "inserted0".into());
+        assert_eq!(node[1], "inserted1".into());
```
Comment (zkat) [inline src/node.rs:684]: this is not what I would expect. [`Vec::insert`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.insert) actually shifts existing elements to the right, so I would expect this assertion to be `.len(), 5`, and for `"pos0"` to be accessible at `node[2]`.

File: src/node.rs [hunk 2]
```diff
@@ -680,4 +668,44 @@ mod test {
...
+        );
+        node.remove(1);
+        assert_eq!(node.entries().len(), 2, "index removal should not succeed");
+        node.remove("not an existing keyword");
+        assert_eq!(node.entries().len(), 2, "key removal should not succeed");
+        node.remove("keyword");
+        assert_eq!(node.entries().len(), 1, "key removal should succeed");
+        node.remove(0);
+        assert_eq!(node.entries().len(), 0, "index removal should suceed");
+        node.remove(0); // should not panic
```
Comment (zkat) [inline src/node.rs:709]: maybe it should panic? That's what [`Vec::remove`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.remove) does, after all.

File: src/node.rs (hunk 1 above)
Comment (jaxter184) [reply src/node.rs:684]: Ah! I see, I misunderstood the doc comment on `KdlNode::insert`. The way you've described seems more intuitive.

File: src/node.rs (hunk 2 above)
Comment (jaxter184) [reply src/node.rs:709]: Seems reasonable to me

File: src/node.rs (hunk 1 above)
Comment (jaxter184) [reply src/node.rs:684]: Oh wait, it looks like the current expected behavior for `KdlNode::insert_impl` is to replace entries, even if using `NodeKey::Index`. In the doc comment, when it says "If an entry already exists with the same key", does "key" in this case mean any `NodeKey` or specifically `NodeKey::Key`?

Comment (zkat) [reply src/node.rs:687]: yeah, it's weird, and imo, it needs to be different depending on whether the key is a number or a property, tbh. I think I had hash keys in mind when I wrote this, but when you're dealing with array-like data, you expect things to shift right.

**Review by zkat — APPROVED** [review]

**Comment by zkat** [comment]
Looks like this just needs a small clippy fix?

**Comment by jaxter184** [comment]
I also squashed a few of the commits together

**Comment by jaxter184** [comment]
third time's the charm

**Comment by zkat** [comment]
woo!

**Pull request merged**